### PR TITLE
Missing changes for v2.6.0 transition

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
## Issue \#

OKTA-414858

## Code

- [x] Unit test(s)
- [x] Implementation

## Additional Context

This was added to the model errata to make the type names consistent:

```js
  { path: 'DNSRecord', rename: 'DnsRecord', renameReason: 'Pattern consistency' },
  { path: 'DNSRecordType', rename: 'DnsRecordType', renameReason: 'Pattern consistency' }, 
```

property errata was extended with the following in order to consume the changed types:
```js
   { path: 'Domain.dnsRecords', model: 'DnsRecord', modelReason: 'Match the changed type name'},
   { path: 'DnsRecord.recordType', model: 'DnsRecordType', modelReason: 'Match the changed type name'},
```

property errata's changes to make properties' names look consistent
```js
   { path: 'PolicyRuleActions.signon', rename: 'SignOn', renameReason: 'Pattern consistency'},
   { path: 'ApplicationSettingsNotes.enduser', rename: 'EndUser', renameReason: 'Pattern consistency'},
```

### CODE REVIEW

File: src/Okta.Sdk/Generated/ApplicationVisibility.Generated.cs [hunk 1]
```diff
@@ -16,13 +16,6 @@ namespace Okta.Sdk
     /// <inheritdoc/>
     public sealed partial class ApplicationVisibility : Resource, IApplicationVisibility
     {
-        /// <inheritdoc/>
-        public bool? AutoLaunch
```
Comment (laura-rodriguez) [inline src/Okta.Sdk/Generated/ApplicationVisibility.Generated.cs:20]: Are we expecting this change? Just curious why this property was added in the previous PR but now we're removing it.

Comment (andriizhegurov-okta) [reply src/Okta.Sdk/Generated/ApplicationVisibility.Generated.cs:20]: Yes! I noticed that too. v2.6.0 spec has changed between those PRs

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,16 @@ Running changelog of releases since `3.1.1`
...
+- Add new models and operations to support the following:
+* [Domains API's endpoints](https://developer.okta.com/docs/reference/api/domains/)
+* [Zones API's endpoints](https://developer.okta.com/docs/reference/api/zones/)
+* [Mappings API's endpoints](https://developer.okta.com/docs/reference/api/mappings/)
+* [ThreatInsight configuration API's endpoints](https://developer.okta.com/docs/reference/api/threat-insight/)
+
+### Bug Fixes
+
+- Fix GroupsClient.ListGroups `filter` parameter doesn't work. Replaced with `search` parameter.
+- Fix AuthorizationServerPolicyRule.ActivateAsync doesn't work.
```
Comment (laura-rodriguez) [inline CHANGELOG.md:18]: Mention that Deactivate and Activate required breaking changes, but since it was a bug fix it's not considered a major change.

Comment (laura-rodriguez) [reply CHANGELOG.md:18]: Add a breaking change section  with details about this:
 public async Task DeleteGroupRuleAsync(string ruleId, bool? removeUsers = null, CancellationToken cancellationToken = default(CancellationToken))

File: src/Okta.Sdk/IGroupsClient.cs [hunk 3]
```diff
@@ -19,5 +19,13 @@ public partial interface IGroupsClient : IAsyncEnumerable<IGroup>
...
         /// <returns>The <see cref="IGroup"/> response.</returns>
         Task<IGroup> CreateGroupAsync(CreateGroupOptions options, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Removes a specific group rule by id from your organization
+        /// </summary>
+        /// <param name="ruleId">Id of the group rule</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A Task that represents the asynchronous operation.</returns>
+        Task DeleteGroupRuleAsync(string ruleId, CancellationToken cancellationToken);
```
Comment (laura-rodriguez) [inline src/Okta.Sdk/IGroupsClient.cs:29]: Should we mark this as obsolete? I have the impression we agreed on that, I might be wrong though.

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,7 @@ Running changelog of releases since `3.1.1`
 * [Zones API's endpoints](https://developer.okta.com/docs/reference/api/zones/)
 * [Mappings API's endpoints](https://developer.okta.com/docs/reference/api/mappings/)
 * [ThreatInsight configuration API's endpoints](https://developer.okta.com/docs/reference/api/threat-insight/)
+- Add an overload for GroupsClient.DeleteGroupRuleAsync function. Now you can specify `removeUsers` parameter indicating whether to keep or remove users from groups assigned by this rule.
```
Comment (laura-rodriguez) [inline CHANGELOG.md:14]: NIt:
Add an overload for `GroupsClient.DeleteGroupRuleAsync` method. Now you can specify the `removeUsers` parameter indicating whether to keep or remove users from groups assigned by this rule.

**Review by laura-rodriguez — APPROVED** [review]
Left 2 minor comments in the last PR. Other than, LGTM 👍 . Thanks for the work here!

File: src/Okta.Sdk/IGroupsClient.cs (hunk 3 above)
Comment (andriizhegurov-okta) [reply src/Okta.Sdk/IGroupsClient.cs:29]: I too remember we were talking about marking it obsolete but then we decided not to do that, that's my impression, but I can't remember why 😕.

Comment (andriizhegurov-okta) [reply src/Okta.Sdk/IGroupsClient.cs:29]: Do you think it's better to mark it obsolete?

**Pull request merged**